app/src/main/java/org/mozilla/focus/browser/integration/BrowserMenuController.kt

```diff
@@ -30,6 +50,25 @@ class BrowserMenuController(
             is ToolbarMenu.Item.Stop -> sessionUseCases.stopLoading(currentTabId)
             is ToolbarMenu.Item.Share -> shareCallback()
             is ToolbarMenu.Item.FindInPage -> showFindInPageCallback()
+            is ToolbarMenu.Item.AddToShortcuts -> {
+                ioScope.launch {
+                    currentTab?.let { state ->
+                        topSitesUseCases.addPinnedSites(
+                            title = state.content.title,
+                            url = state.content.url
+                        )
+                    }
+                }
+            }
+            is ToolbarMenu.Item.RemoveFromShortcuts -> {
+                ioScope.launch {
+                    currentTab?.let { state ->
+                        topSitesStorage.findTopSite(url = state.content.url)?.let { topSite ->
+                            topSitesUseCases.removeTopSites(topSite)
+                        }
+                    }
+                }
+            }
```

Comment:
These 2 should also trigger a state change. Otherwise, you will still have the same option in the menu for a site you already added.

Reply:
The DefaultTopSiteStorage will observe these changes and notify the observers, which will eventually get to our view that will dispatch the state changes.

Reply:
I was talking about the menu that stays the same `hasTopSite` and `canAddTopSite` do not change after these actions.

Reply:
I did investigate this, but I don't think it has to do with dispatching the states but rather menu items are cached. This can be a follow up.